ecms_base/modules/custom/ecms_blocks/src/Form/PublicationAudienceForm.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\ecms_blocks\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+
+/**
+ * Provides a publication audience form.
+ *
+ * @package Drupal\ecms_blocks\Form
+ */
+class PublicationAudienceForm extends FormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId(): string {
+    return 'ecms_publication_audience_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+
+    // Load audience values.
+    $vid = 'publication_audience';
```

Comment:
It might be better to define the `$vid` as a class constant instead. 
`const VID = 'publication_audience';`

And then use it with `self::VID`.